src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserBadgeManager.java

```diff
@@ -0,0 +1,154 @@
+import uk.ac.cam.cl.dtg.segue.dao.userBadges.teacherBadges.TeacherGroupsBadgePolicy;
+import uk.ac.cam.cl.dtg.segue.dos.UserBadge;
+import uk.ac.cam.cl.dtg.segue.dao.userBadges.IUserBadgePolicy;
+import uk.ac.cam.cl.dtg.segue.dto.users.RegisteredUserDTO;
+
+import java.sql.Connection;
+import java.util.List;
+import java.util.Map;
+
+import static uk.ac.cam.cl.dtg.segue.api.Constants.CONTENT_INDEX;
+
+/**
+ * Management layer for updating and retrieving badge states
+ *
+ * Created by du220 on 27/04/2018.
+ */
+public class UserBadgeManager {
+
+    public enum Badge {
+        // teacher specific badges
+        TEACHER_GROUPS_CREATED,
+        TEACHER_ASSIGNMENTS_SET,
+        //TEACHER_BOOK_PAGES_SET,
+        TEACHER_GAMEBOARDS_CREATED,
+        TEACHER_CPD_EVENTS_ATTENDED
+    }
+
+    private final IUserBadgePersistenceManager userBadgePersistenceManager;
+    private final Map<Badge, IUserBadgePolicy> badgePolicies = Maps.newHashMap();
+
+    /**
+     * Constructor
+     *
+     * @param userBadgePersistenceManager badge persistence object for liasing with database
+     * @param groupManager group manager object for badge policy dependencies
+     * @param bookingManager booking manager object for badge policy dependencies
+     * @param assignmentManager assignment manager object for badge policy dependencies
+     * @param gameManager game manager object for badge policy dependencies
+     * @param contentManager content manager object for badge policy dependencies
+     * @param contentIndex specifies content version
+     */
+    @Inject
+    public UserBadgeManager(IUserBadgePersistenceManager userBadgePersistenceManager, GroupManager groupManager,
+                            EventBookingManager bookingManager, AssignmentManager assignmentManager, GameManager gameManager,
+                            IContentManager contentManager, @Named(CONTENT_INDEX) String contentIndex) {
+
+        this.userBadgePersistenceManager = userBadgePersistenceManager;
+
+        badgePolicies.put(Badge.TEACHER_GROUPS_CREATED, new TeacherGroupsBadgePolicy(groupManager));
+        badgePolicies.put(Badge.TEACHER_ASSIGNMENTS_SET, new TeacherAssignmentsBadgePolicy(assignmentManager,
+                gameManager));
+        badgePolicies.put(Badge.TEACHER_GAMEBOARDS_CREATED, new TeacherGameboardsBadgePolicy(gameManager));
+        badgePolicies.put(Badge.TEACHER_CPD_EVENTS_ATTENDED,
+                new TeacherCpdBadgePolicy(bookingManager, contentManager, contentIndex));
+    }
+
+    /**
+     * Gets an up-to-date badge by either retrieving from the database or initialising first-time on the fly
+     *
+     * @param conn database connection
+     * @param user owner of badge record
+     * @param badgeName enum of badge to be updated
+     * @return user badge object
+     * @throws SegueDatabaseException
+     */
+    public UserBadge getOrCreateBadge(Connection conn, RegisteredUserDTO user, Badge badgeName)
```

Comment:
Could we avoid the manager level having to know about sql classes e.g. Connection? I would normally expect that to be hidden away in the DAO layer. It is not urgent but maybe something to refactor later?